### DISCUSSION TITLE
Unable to delete cookie using Next.js server side action

### DISCUSSION THREAD

**Discussion by riculum** [vercel/next.js#51621] [Help]
### Summary

I'm trying to delete a cookie using the next/headers module in a Next.js application, but it doesn't seem to work as expected. Here's the code snippet:

```
import {cookies} from "next/headers";
export default async function Signout() {
    async function deleteTokens() {
       "use server"

        cookies().delete('accessToken')
    }

  await deleteTokens()
  return (
      <></>
  );
}
```

I expected the cookies().delete('accessToken') line to delete the cookie named "accessToken", but it doesn't seem to have any effect. The cookie is still present after the function is executed.

I am getting following error: Error: Cookies can only be modified in a Server Action or Route Handler. Read more: https://nextjs.org/docs/app/api-reference/functions/cookies#cookiessetname-value-options

But i have set the "use server" and enabled it in the next.config.js

#using Next.js 13.4.4

**Comment by sanny-io** [comment]
I also had this issue and just fixed it. My cookie was set for `.mydomain.com` (note the period in front to share the cookie with subdomains).

So the same domain must also be specified when deleting.

```javascript
cookieStore.delete({
    name: 'accessToken',
    domain: `.${new URL(process.env.NEXT_PUBLIC_BASE_URL).hostname}`,  // http://mydomain.com --> .mydomain.com
})
```
